core/index/src/main/java/mil/nga/giat/geowave/core/index/ByteArrayRange.java

```diff
@@ -108,26 +109,28 @@ public int compareTo(
 				other.getEnd());
 	}
 
-	public static final void mergeIntersections(
+	public static final List<ByteArrayRange> mergeIntersections(
 			List<ByteArrayRange> ranges,
 			int maxRanges ) {
 		// sort order so the first range can consume following ranges
 		Collections.<ByteArrayRange> sort(ranges);
-		// merge in place
-		for (int i = 0; i < ranges.size(); i++) {
+		final List<ByteArrayRange> result = new ArrayList<ByteArrayRange>();
+		for (int i = 0; i < ranges.size(); ) {
 			ByteArrayRange r1 = ranges.get(i);
-			for (int j = i + 1; j < ranges.size(); j++) {
+			int j = i + 1;
+			for (; j < ranges.size(); j++) {
 				final ByteArrayRange r2 = ranges.get(j);
 				if (r1.intersects(r2)) {
 					r1 = r1.merge(r2);
-					ranges.remove(j);
-					j--;
-					ranges.set(
-							i,
-							r1);
+				}
+				else {
+					break;
 				}
 			}
+			i = j;
```

Comment:
logically this seems to have a potential inconsistency, right?  let's say you have ranges, a,b,c, and d (starts of the range are sorted in that order).  Imagine a, b, and c are very long spans, d is a very short span.  a is so long it overlaps d, but b and c aren't quite as long but overlap each other.  It seems this logic would find that a and d overlap, and skip merging b and c together.  I know this may be an edge case that we're willing to trade performance for algorithmic accuracy here.  I just wanted to at least point it out, so if we decide the accuracy tradeoff is still worth it, we at least add a comment to explain.